services/decision/controller/controller.go

```diff
@@ -147,6 +147,12 @@ func FinalizeDecision(w http.ResponseWriter, r *http.Request) {
 		if err != nil {
 			panic(errors.InternalError(err.Error(), "Could not add user to mail list."))
 		}
+	} else if !updated_decision.Finalized {
+		err = service.RemoveUserFromMailList(id, updated_decision)
```

Comment:
Shouldn't we be removed from the list associated with the original decision not their updated decision?

Reply:
As far as I understand, `FinalizeDecision` only changes the `Finalized` status. By this time, the actual decision shouldn't have changed from accepted -> denied etc..

Reply:
Yes this is fair.